Window size and maximization is now preserved between runs of Pithos

ChaiTRex opened the pull request:
While Pithos remembers the position of the upper-left corner of the Pithos window in between runs of Pithos, it doesn't remember the size of the window, which makes it where I have to manually resize it every time I run it.

This pull request causes Pithos to remember the size of the window between runs.

JasonLG1979 commented:
You should probably put that in the UI file and use the @GtkTemplate.

JasonLG1979 commented on pithos/pithos.py:
```diff
@@ -1128,6 +1133,10 @@ def on_configure_event(self, widget, event, data=None):
         self.settings.set_value('win-pos', GLib.Variant('(ii)', (event.x, event.y)))
         return False
 
```
@GtkTemplate.Callback

JasonLG1979 replied:
And connect the signal handler in https://github.com/pithos/pithos/blob/master/data/ui/PithosWindow.ui

JasonLG1979 replied:
Under
https://github.com/ChaiTRex/pithos/blob/master/data/ui/PithosWindow.ui#L16

Put:
```xml
    <signal name="check-resize" handler="on_window_resize" swapped="no"/>
```

TingPing replied:
This should be in the `configure-event` callback just like the position is.

JasonLG1979 replied:
Easy enough I was just going off what signal they were using. That simplifies things.

JasonLG1979 commented on pithos/pithos.py:
```diff
@@ -127,6 +127,8 @@ def __init__(self, app, test_mode):
         super().__init__(application=app)
         self.init_template()
 
+        self.connect('check-resize', self.on_window_resize)
```
Get rid of this.

ChaiTRex commented:
OK, I've:

0. Added `@GtkTemplate.callback`
0. Moved the signal handler connection to `PithosWindow.ui`
0. Added maximization support, since maximizing and then quitting was screwing things up (window would forget its unmaximized size and be as large as it's possible to be without maximization)

I'm new to Python and GTK, so I'm not sure how to combine everything into one callback like @TingPing suggested.

JasonLG1979 commented:
Sorry scrap what a said. @TingPing wants everything that saves settings in the configure-event handler. Also you wouldn't want to try to resize the window every time you brought it to the front. In my experience every window manager I've ever used has no problem remembering the size of a running window. Resizing the window would be a one shot at start up deal.

What you're looking for is something like this:

https://github.com/JasonLG1979/pithos/commit/b335b0051182064b09f6ce7ebd4fdf9018ee7f03

TingPing commented:
Slightly off topic but should Pithos even allow being maxmized? The UI was clearly not designed for it.

JasonLG1979 commented:
@TingPing I agree 100% but I don't think there's a way to disallow maximizing that all window managers follow. I've messed with it in Tk and depending on the WM I got inconsistent behavior. Could have just been Tk it's a pretty basic toolkit?

TingPing commented:
@JasonLG1979 Looks like `gtk_window_set_resizable()` was what I was thinking of which is not quite what we want.

JasonLG1979 commented:
@TingPing It would also be really easy to remember if the window was minimized when Pithos closed. Just make the 'min-max' a string and set the value to 'min', max' or '' or make min-max an int and use 0 - 2 to represent min, normal, max. But I'm not sure about starting Pithos minimized. It could be confusing and like you said the Pithos isn't really made to be maximized. I'm not even sure I like remembering the state at all. I still like the idea of keeping track of if Pithos is maximized for the purpose of position and size so it can properly be ignored though.

TingPing commented:
> But I'm not sure about starting Pithos minimized. It could be confusing

My thoughts exactly.

JasonLG1979 commented:
It might be a good pref option but not something you'd want to do by default. I don't know about you but most of the time Pithos isn't focused.

JasonLG1979 commented:
@TingPing your thoughts on even remembering state?

JasonLG1979 commented:
This is almost a year old now with no activity and it's questionable if it's even very useful. I'm going to go a head and close this. @TingPing or @ChaiTRex you guys can re-open it if you'd like?
